spec/features/user/user_can_invite_to_app_spec.rb

```diff
@@ -0,0 +1,124 @@
+require 'rails_helper'
+
+RSpec.describe 'User can send an email invite to the application' do
+  describe 'As a logged in user, my dashboard' do
+    before :each do
+      @user = create(:user)
+
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@user)
+
+      stub_user_1_dashboard
+
+      ActionMailer::Base.deliveries = []
+
+    end
+    describe 'if I am connected to Github' do
+      before :each do
+        create(:github_token, user: @user, token: ENV['USER_1_GITHUB_TOKEN'])
+      end
+
+      it 'has a Send an Invite button' do
+
+        visit dashboard_path
+
+        click_button("Send an Invite")
+
+        expect(current_path).to eq(invite_path)
+
+      end
+
+      describe 'When I click on send an Invite'  do
+        it 'my path is now /invite' do
+
+          visit dashboard_path
+
+          click_button("Send an Invite")
+
+          expect(current_path).to eq('/invite')
+
+        end
+
+        describe 'if I enter a valid github handle' do
+          it 'should send an email to that github user inviting them to the app' do
+            visit dashboard_path
+
+            click_button("Send an Invite")
+
+            expect(current_path).to eq('/invite')
+
+            fill_in 'handle', with: 'teresa-m-knowles'
```

Comment:
Wonder if we should prevent inviting yourself. In fact, should we stop the sending an invite to anyone already signed up...? 🙈

Reply:
Yes, that's an excellent edge case. I'll do something like comparing the current_user.uid with the github_user.id and then add an error message for that.

Maybe if the github_user's id is already a uid in our database we can let the user know. Something like "#{github_user.name} is already registered in Brownfield of Dreams!"

Reply:
Yep, thats what I was thinkin.